File: src/Client/HCI/MainDialog/AssetsPanel/ToolbarItemData.py
#  Copyright 2014-2016 Integrated Test Management Centre Foundation Team
#  Licensed under the Apache License, Version 2.0 (the "License");
#  you may not use this file except in compliance with the License.
#  You may obtain a copy of the License at
#  http://www.apache.org/licenses/LICENSE-2.0
#  //////////////////////////////////////////////////////////////////////////
import wx


#include <wx/treectrl.h>


class ToolbarItemSubType(object):
	Unknown = 0
	TestingAssetsFilter = 1
	TestingAssetsTestItem = 2
	TestingAssetsTestGroup = 3


class ToolbarItemData(wx.TreeItemData):

	def __init__(self,id, subType, name):
		self._itemID = id
		self._subType = subType
		self._name = name


	def GetID(self): return self._itemID

	def GetNodeSubType(self): return self._subType

	def GetName(self): return self._name

File: sourcecode/client/src/hci/mainDialog/assets/toolbarItemData.cpp
//  Copyright 2014 Integrated Test Management Centre Foundation Team
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
// http://www.apache.org/licenses/LICENSE-2.0
//////////////////////////////////////////////////////////////////////////
#include <hci/mainDialog/assets/toolbarItemData.h>


ToolbarItemData::ToolbarItemData( int id
                                , ToolbarItemSubType subType
                                , std::string name )
                                : m_itemID( id )
                                , m_subType( subType )
                                , m_name( name )
{
}


ToolbarItemData::~ToolbarItemData()
{
}
